Source/RapyutaSimulationPlugins/Public/Core/RRNetworkPlayerController.h

```diff
@@ -31,70 +32,46 @@ class RAPYUTASIMULATIONPLUGINS_API ARRNetworkPlayerController : public APlayerCo
     void GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const override;
 
     // ROS2
-    UPROPERTY(Transient, Replicated)
+    UPROPERTY(Transient)
     AROS2Node* SimStateClientROS2Node = nullptr;
 
-    UPROPERTY(BlueprintReadOnly, Replicated)
+    UPROPERTY(BlueprintReadOnly)
     URRROS2ClockPublisher* SimStateClientClockPublisher = nullptr;
 
-    /**
-     * @brief Init SimStateClient's ROS2 (Node, ClockPublisher, etc.)
-     */
-    void InitSimStateClientROS2();
-
     // SIM STATE & ROS2 STATE CLIENT
     //! Pointer to ServerSimState
     UPROPERTY(EditAnywhere, BlueprintReadWrite, meta = (ExposeOnSpawn = "true"), Replicated)
     ASimulationState* ServerSimState = nullptr;
+
     //! ROS2 Sim state client
-    UPROPERTY(Transient, Replicated)
+    UPROPERTY(Transient, Replicated, ReplicatedUsing = OnRep_SimStateClient)
     URRROS2SimulationStateClient* ROS2SimStateClient = nullptr;
-    /**
-     * @brief Create ROS2SimStateClient without initializing yet
-     */
-    void CreateROS2SimStateClient(const TSubclassOf<URRROS2SimulationStateClient>& InSimStateClientClass);
 
-    // POSSESSED PAWN
-    //! Pawn that has been possessed by this controller
-    UPROPERTY(Transient, Replicated)
-    APawn* PossessedPawn = nullptr;
+    UFUNCTION(BlueprintCallable)
+    virtual void OnRep_SimStateClient();
 
-    /**
-     * @brief Search for the targeted pawn to possess
-     */
-    APawn* FindPawnToPossess();
+    // PLAYER
+    //! Controller's PlayerName, taken from [robotname] param in Engine build.
+    //! In Editor build, it will be automatically set from #PossessedPawn's name upon possession
+    UPROPERTY(EditAnywhere, BlueprintReadWrite, meta = (ExposeOnSpawn = "true"), Replicated)
+    FString PlayerName;
 
     /**
-     * @brief Wait to possess the targeted pawn
+     * @brief Create ROS2SimStateClient without initializing yet
      */
-    UFUNCTION()
-    void WaitToPossessPawn();
+    UFUNCTION(BlueprintCallable)
+    void CreateROS2SimStateClient(const TSubclassOf<URRROS2SimulationStateClient>& InSimStateClientClass);
 
     /**
-     * @brief Possess a pawn on server
+     * @brief Init SimStateClient's ROS2 (Node, ClockPublisher, etc.)
      */
-    UFUNCTION(Server, Reliable)
-    void ServerPossessPawn(APawn* InPawn);
-
-    //! Timer handle used by WaitToPossessPawn()
-    UPROPERTY(VisibleAnywhere, BlueprintReadOnly)
-    FTimerHandle PossessTimerHandle;
-
-    //! Client - Init possessed robot's ROS2Interface & MoveComponent, etc.
     UFUNCTION(Client, Reliable)
-    void ClientInitPawn(AActor* InActor);
-
-    // PLAYER
-    //! Controller's PlayerName, taken from [robotname] param in Engine build.
-    //! In Editor build, it will be automatically set from #PossessedPawn's name upon possession
-    UPROPERTY(EditAnywhere, BlueprintReadWrite, meta = (ExposeOnSpawn = "true"), Replicated)
-    FString PlayerName;
+    void InitSimStateClientROS2();
```

Comment:
```suggestion
    void ClientInitSimStateClientROS2();
```
with `Client/Server` prefix as a [convention](https://docs.unrealengine.com/4.26/en-US/InteractiveExperiences/Networking/Actors/RPCs/#quicktips) for an RPC method